docusaurus/docs/iOS/components/message-list.md

```diff
@@ -1,3 +1,266 @@
+            appearance: appearance
+        )
+    }
+}
+```
+As you can see, we implement the data source to feed the message list with the messages from the `ChannelController`. The most complex part is to calculate the layout options for each message, but since our custom `ChannelViewController` implements the `ThemeProvider` protocol, we have access to the `components` and `appearance` config, and with these we can use the `MessageLayoutOptionsResolver` to calculate the layout options for us.
+
+The `ChatMessageListVC` provides events throught the `ChatMessageListVCDelegate` so that we can react to actions and important lifecycle events.
+```swift
+extension ChannelViewController: ChatMessageListVCDelegate {
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        willDisplayMessageAt indexPath: IndexPath
+    ) {
+        // Load previous messages
+
+        if channelController.state != .remoteDataFetched {
+            return
+        }
+
+        if indexPath.row < channelController.messages.count - 10 {
+            return
+        }
+
+        channelController.loadPreviousMessages()
+    }
+
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        didTapOnAction actionItem: ChatMessageActionItem,
+        for message: ChatMessage
+    ) {
+        // Handle message actions
+        switch actionItem {
+        case is EditActionItem:
+            dismiss(animated: true) { [weak self] in
+                self?.messageComposerVC.content.editMessage(message)
+            }
+        case is InlineReplyActionItem:
+            dismiss(animated: true) { [weak self] in
+                self?.messageComposerVC.content.quoteMessage(message)
+            }
+        case is ThreadReplyActionItem:
+            dismiss(animated: true) { [weak self] in
+                self?.messageListVC.showThread(messageId: message.id)
+            }
+        default:
+            return
+        }
+    }
+
+    open func chatMessageListVC(_ vc: ChatMessageListVC, scrollViewDidScroll scrollView: UIScrollView) {
+        // handle scroll events, and check if the last message was read, to mark the channel read.
+        if messageListVC.listView.isLastCellFullyVisible, channelController.channel?.isUnread == true {
+            channelController.markRead()
+        }
+    }
+
+}
+```
+Currently, by implementing the `ChatMessageListVCDelegate` we are able to handle when a user performs an action on a message, when a message will be displayed, and when the user is scrolling the message list. More events might be added in the future, but for now these should be enough to implement the most common features in a chat view, like pagination, marking the channel as read when the user scrolls to the bottom, and handling message actions.
+
+Finally, we need to implement the `ChannelControllerDelegate` to handle the events from the `ChannelController`. This will make sure that the messages are always in sync with the server.
```

Comment:
```suggestion
Finally, we need to implement the `ChannelControllerDelegate` to handle the events from the `ChannelController`. This will make sure the message are always in sync with the server.
```